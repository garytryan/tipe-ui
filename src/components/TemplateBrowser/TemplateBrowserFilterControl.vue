<template>
  <div :data-tipe-ui="$options.name">
    <button
      v-for="({ label, value }, i) in filters"
      :key="i"
      :class="buttonClassObject(value)"
      @click="() => onClick(value)"
    >
      {{ label }}
    </button>
  </div>
</template>

<script>
const filters = [
  { label: 'Recently Viewed', value: 'recently-viewed' },
  { label: 'Starred', value: 'starred' },
  { label: 'Created by me', value: 'created-by-me' },
  { label: 'Shared with me', value: 'shared-with-me' }
]

export default {
  name: 'TipeTemplateBrowserFilterControl',
  data() {
    return {
      value: 'recently-viewed',
      filters
    }
  },
  methods: {
    buttonClassObject(value) {
      return {
        on: value === this.value
      }
    },
    onClick(value) {
      this.value = value
      this.$emit('change', value)
    }
  }
}
</script>

<style lang="postcss" scoped>
[data-tipe-ui='TipeTemplateBrowserFilterControl'] {
  height: auto;
  width: auto;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: flex-start;
}

button {
  height: auto;
  width: auto;
  border: none;
  outline: none;
  padding: 0 1rem 0 0;
  margin: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 0.8125rem;
  background-color: transparent;
  white-space: nowrap;
}

button.on {
  font-weight: 800;
}
</style>
