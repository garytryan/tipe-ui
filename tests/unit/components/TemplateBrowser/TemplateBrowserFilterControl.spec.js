import { shallowMount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import TemplateBrowserFilterControl from '@/components/TemplateBrowser/TemplateBrowserFilterControl'

describe('TemplateBrowserFilterControl', () => {
  describe('<TemplateBrowserFilterControl>', () => {
    it('matches previous snapshot', () => {
      const renderer = createRenderer()
      const wrapper = shallowMount(TemplateBrowserFilterControl)

      renderer.renderToString(wrapper.vm, (err, str) => {
        if (err) throw new Error(err)
        expect(str).toMatchSnapshot()
      })
    })

    it('has correct data-tipe-iu attibute', () => {
      const wrapper = shallowMount(TemplateBrowserFilterControl)

      expect(wrapper.attributes()['data-tipe-ui']).toBe(
        'TipeTemplateBrowserFilterControl'
      )
    })

    it('<button> - first button has "on" class by default', () => {
      const wrapper = shallowMount(TemplateBrowserFilterControl)

      expect(wrapper.find('button').classes()).toContain('on')
    })
  })

  describe('@events', () => {
    it('@change - emitted when button clicked', () => {
      const wrapper = shallowMount(TemplateBrowserFilterControl)

      wrapper.find('button').trigger('click')
      expect(wrapper.emitted().change).toBeTruthy()
    })

    it('@click - button has "on" class after clicked', () => {
      const wrapper = shallowMount(TemplateBrowserFilterControl)
      const secondButton = wrapper.findAll('button').at(1)

      expect(secondButton.classes()).not.toContain('on')
      secondButton.trigger('click')
      expect(secondButton.classes()).toContain('on')
    })
  })
})
