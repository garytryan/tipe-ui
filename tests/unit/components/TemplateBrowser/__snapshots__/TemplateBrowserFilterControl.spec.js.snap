// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemplateBrowserFilterControl <TemplateBrowserFilterControl> matches previous snapshot 1`] = `
<div data-tipe-ui="TipeTemplateBrowserFilterControl">
  <button class="on">
    Recently Viewed
  </button>
  <button>
    Starred
  </button>
  <button>
    Created by me
  </button>
</div>
`;
